components/buildless-serverless/api/v1alpha2/function_types.go

```diff
@@ -198,12 +198,20 @@ type FunctionStatus struct {
 	FunctionResourceProfile string `json:"functionResourceProfile,omitempty"`
 	// Specifies an array of conditions describing the status of the parser.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
-	// Specifies the commit hash used to build the Function.
+	// Deprecated: Specifies the commit hash used to build the Function.
+	// Field moved to GitRepositoryStatus.
 	Commit string `json:"commit,omitempty"`
-	// Specify the repository which was used to build the function.
+	// Deprecated: Specify the repository which was used to build the function.
+	// Field moved to GitRepositoryStatus.
 	Repository `json:",inline,omitempty"`
+	// Specifies the GitRepository status when the Function is sourced from a Git repository.
+	GitRepository *GitRepositoryStatus `json:"gitRepository,omitempty"`
+}
 
-	//TODO Should we add the GitRepository URL here?
+type GitRepositoryStatus struct {
```

Comment:
Should we add also the `reference` field?

Reply:
We have it inside `Repository` struct.

Reply:
Yep, my oversight, thanks